src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,56 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { Status } from '../../types/Status';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  inputValue: string;
+  handleSelectedStatus: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+  handleQueryChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  resetStatus: () => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = ({
+  handleQueryChange,
+  handleSelectedStatus,
+  inputValue,
+  resetStatus,
+}: Props) => {
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select data-cy="statusSelect" onChange={handleSelectedStatus}>
+            <option value="all">{Status.ALL}</option>
+            <option value="active">{Status.ACTIVE}</option>
+            <option value="completed">{Status.COMPLETED}</option>
```

Comment:
The `value` attributes for the `option` elements should match the `Status` enum values. Currently, they are hardcoded as strings ('all', 'active', 'completed'), which may not align with the `Status` enum values. Consider using `Status.ALL`, `Status.ACTIVE`, and `Status.COMPLETED` for consistency and to avoid potential mismatches.